### PR TITLE
Aida 378

### PR DESCRIPTION
**Pull request by bdpham**
added missing code from merge

### CODE REVIEW

File: src/app/components/network-graph/network-graph.component.ts [hunk 1]
```diff
@@ -1101,29 +1085,47 @@ export class NetworkGraphComponent extends BaseNeonComponent implements OnInit,
...
-            let field = selectedNode.isLink ? this.options.linkField : this.options.nodeField;
-            let myFilter = {
-                id: undefined,
-                field: field.columnName,
-                prettyField: field.prettyName,
-                value: nodeName,
-                operator: '='
-            };
-            this.addFilter(myFilter);
+            for (let filterField of selectedNode.filterFields) {
```
Comment (ThomasSchellenbergNextCentury) [inline src/app/components/network-graph/network-graph.component.ts:1112]: Calling addNeonFilter within a loop (by calling addFilter on lines 1105 and 1111) will cause errors due to race conditions.  You need to call addNeonFilter once on your first filter, wait for a successful response from the backend, and then call addNeonFilter again on your second filter.  Right now, you're calling addNeonFilter in rapid-fire succession, which sends all the add-filter requests to the backend without waiting for the backend to process the requests and thus may get the backend filter state out-of-whack.

The addNeonFilter function has a 4th argument for a callback function.  You need to move the loop behavior to a separate function and then, on your first call to addNeonFilter supply a callback that calls your new function to add the next filter.  Let me know if you have questions or I'm not being clear.

**Review by ThomasSchellenbergNextCentury — CHANGES_REQUESTED** [review]
You've got an issue with your filter-adding behavior.  Otherwise everything looks fine.

File: src/app/components/network-graph/network-graph.component.ts (hunk 1 above)
Comment (bdpham) [reply src/app/components/network-graph/network-graph.component.ts:1112]: Ahhh that makes sense. I will fix that. Thanks.

**Review by gitnordt — CHANGES_REQUESTED** [review]
I noticed that if you double-click on a node in the Network Graph to filter and then click on a hypothesis, the closeable filter in the network graph disappears as if the filter was removed. However if you look at the filter tray, the filter still exists. 
screenshot from 2018-10-12 13-56-07

**Comment by bdpham** [comment]
@snorpet I will put in a different issue for the bug that you found.

**Review by ThomasSchellenbergNextCentury — APPROVED** [review]

**Review by gitnordt — COMMENTED** [review]
That's acceptable. Let me know the bug number for the filtering issue.

**Pull request merged**